Sources/DistributedActors/Cluster/ClusterShell.swift

```diff
@@ -625,16 +635,24 @@ extension ClusterShell {
             context.system.cluster.updateMembershipSnapshot(state.membership)
 
             // Publish the events
+            var eventsToPublish: [Cluster.Event] = []
             mergeDirective.effectiveChanges.forEach { effectiveChange in
                 // a change COULD have also been a replacement, in which case we need to publish it as well the removal od the
                 if let replacementChange = effectiveChange.replacementDownPreviousNodeChange {
-                    self.clusterEvents.publish(.membershipChange(replacementChange))
+                    eventsToPublish.append(.membershipChange(replacementChange))
                     self.tryConfirmDeadToSWIM(context, state, change: replacementChange)
                 }
-                self.clusterEvents.publish(.membershipChange(effectiveChange))
+                eventsToPublish.append(.membershipChange(effectiveChange))
                 self.tryConfirmDeadToSWIM(context, state, change: effectiveChange)
             }
 
+            let events = eventsToPublish
+            Task {
```

Comment:
Was this causing sendable warnings?  Maybe this can be `Task { [eventsToPublish]` if so?

Reply:
amended https://github.com/apple/swift-distributed-actors/pull/1012/commits/f80caa21f4a8a3d54b9c5926b3bd3df0a4ca2b76